Implemented BinarySearch.py

gennady9 opened the pull request:
https://github.com/ekdnam/Algorithms/issues/39

raniphore commented on Searching/Binary Search/BinarySearch.py:
```diff
@@ -0,0 +1,28 @@
+
+# Printing
+found = (result != -1)
+found_message = ("Found at index " + str(result)) if found else "Not found"
```
We can directly check the value of result to frame the found_message, no need to assign it to a boolean variable found.

ekdnam replied:
Yes, that can be done

ekdnam commented:
Add the functionality of taking the input of the graph from the user, rather than statically, and I will merge it!

ekdnam commented:
Good job! Add the functionality of taking the input of the tuple from the user, rather than statically, and I will merge it!

gennady9 commented:
Thanks :)

When you say "taking the input from the user", do you mean using input() function?
I'm not sure why would we want that, I thought the purpose was to create a function..
the static array is there just for testing purposes

Anyway, I can do that - I can do a while loop with input() in it, that stops with a certain number/string
I just need you to define that stopping condition

ekdnam commented:
The purpose of this repo in my mind is, that, if we anyone clones this repo, they can check the functions with different input values, and try out test cases, find out edge cases. Thus the user input.

ekdnam commented:
> Anyway, I can do that - I can do a while loop with input() in it, that stops with a certain number/string
I just need you to define that stopping condition

Let the user enter the number of array elements (n), and then the while loop takes input of n tuple elements. And then, they enter the element to be searched, and then the search function is called with the required parameters

gennady9 commented:
> > Anyway, I can do that - I can do a while loop with input() in it, that stops with a certain number/string
> > I just need you to define that stopping condition
> 
> Let the user enter the number of array elements (n), and then the while loop takes input of n tuple elements. And then, they enter the element to be searched, and then the search function is called with the required parameters

I added the changes, and also changed the name to RecursiveBinarySearch.py in PR #123 (https://github.com/ekdnam/Algorithms/pull/123)
